app/ui/legacy/src/main/java/com/fsck/k9/ui/push/PushInfoFragment.kt

```diff
@@ -37,19 +39,21 @@ class PushInfoFragment : Fragment() {
 
     private fun initializeNotificationSection(view: View) {
         val notificationTextView = view.findViewById<MaterialTextView>(R.id.notificationText)
+        val notificationExplanationText =
+            getString(R.string.push_info_notification_explanation_text, appNameProvider.appName)
+        val notificationConfigureText = getString(R.string.push_info_configure_notification_text)
+
         if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
-            val configureNotificationText = getString(R.string.push_info_configure_notification_text)
-            notificationTextView.text = getString(
-                R.string.push_info_notification_explanation_text,
-                configureNotificationText,
-            )
+            val notificationText = "$notificationExplanationText $notificationConfigureText"
```

Comment:
We should avoid concatenating user-visible strings in code. Assumptions based on English don't necessarily hold in other languages. In this case the Chinese translation doesn't use a space between the two sentences.

Changing the string resource to…

```xml
<string name="push_info_notification_explanation_text">When using Push, <xliff:g id="app_name">%1$s</xliff:g> maintains a connection to the mail server. Android requires displaying an ongoing notification while the app is active in the background. %2$s</string>
```
…and keeping the existing code structure gives the necessary flexibility to translators.

```kotlin
if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
    val configureNotificationText = getString(R.string.push_info_configure_notification_text)
    notificationTextView.text = getString(
        R.string.push_info_notification_explanation_text,
        appNameProvider.appName,
        configureNotificationText,
    )
    val configureNotificationButton = view.findViewById<MaterialButton>(R.id.configureNotificationButton)
    configureNotificationButton.isVisible = true
    configureNotificationButton.setOnClickListener {
        launchNotificationSettings()
    }
} else {
    notificationTextView.text = getString(
        R.string.push_info_notification_explanation_text,
        appNameProvider.appName,
        "",
    )
}
```